music21/osmd/osmd.py

```diff
@@ -0,0 +1,222 @@
+from music21.instrument import Piano
+from music21.common import getSourceFilePath
+from music21.common import runningUnderIPython
+from music21 import exceptions21
+
+
+class OpenSheetMusicDisplayException(exceptions21.Music21Exception):
+    pass
+
+
+try:
+    loader = importlib.util.find_spec('IPython.core.display')
+except ImportError:
+    loader = None
+hasInstalledIPython = loader is not None
+del importlib
+
+
+def getExtendedModules():
+    # pylint: disable=import-outside-toplevel
+    if hasInstalledIPython:
+        from IPython.core.display import display, HTML, Javascript
+    else:
+        def display():
+            raise OpenSheetMusicDisplayException(
+                'OpenSheetMusicDisplay requires IPython to be installed')
+
+        HTML = Javascript = display
+    return display, HTML, Javascript
+
+
+class ConverterOpenSheetMusicDisplay(SubConverter):
+    '''
+    Converter for displaying sheet music inline
+    in an IPython notebook.
+    Uses: https://opensheetmusicdisplay.org/ for rendering
+    '''
+    registerFormats = ('osmd',)
+    registerShowFormats = ('osmd',)
+    # when updating the script tag osmd will only reload in a notebook if you: Kernel -> restart and
+    # clear output, then save the notebook and refresh the page.
+    # Otherwise the script will stay on the page and not reload.
+    script_url = ("https://github.com/opensheetmusicdisplay"
+                  + "/opensheetmusicdisplay/releases/download/0.9.2/opensheetmusicdisplay.min.js")
+    osmd_file = os.path.join(getSourceFilePath(), 'osmd', 'opensheetmusicdisplay.0.9.2.min.js')
+
+    def __init__(self):
+        super().__init__()
+        self.display, self.HTML, self.Javascript = getExtendedModules()
+
+    def show(self, obj, fmt,
+             fixPartName=True, offline=False, divId=None,
+             **keywords):
+        r'''
+        Displays the score object in a notebook using the OpenSheetMusicDisplay.js library.
+
+        >>> import music21
+        >>> s = music21.converter.parse("tinyNotation: 3/4 E4 r f# g=lastG trip{b-8 a g} c4~ c")
+        >>> fig_id1 = s.show('osmd')
+
+        To update a previously displayed score use the returned divId from before:
+
+        >>> fig_id2 = s.show('osmd', divId=fig_id1)
+        >>> assert fig_id1 == fig_id2
+        '''
+        in_ipython = runningUnderIPython()
+        score = obj
+        if fixPartName:
+            self.addDefaultPartName(score)
+
+        if divId is None:
+            # create unique reference to output div in case we wish to update it
+            divId = self.getUniqueDivId()
+
+
+        xml = open(score.write('musicxml')).read()
+        script = self.musicXMLToScript(xml, divId, offline=offline)
+        if in_ipython:
+            self.display(self.HTML(f'<div id="{divId}">loading OpenSheetMusicDisplay</div>'))
+            self.display(self.Javascript(script))
+        else:
+
+            tmp = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
For consistency you can use `environLocal.getTempFile()` so that all the music21 temp files are in the same place. 
Most files have something like this at the top:
```
from music21 import environment
_MOD = 'osmd'
environLocal = environment.Environment(_MOD)
```

Reply:
yes, please!

Though I'm in the `_MOD` killing mood, so just go ahead and do:

`environLocal = environment.Environment('osmd.osmd')`

(the _MOD should be the path after `music21.`)